### PR TITLE
Add HTTP attributes

### LINKED ISSUE #385: Add attributes to HTTP spans
Currently, attributes such as HTTP URL, method, request size, response size and error status are not automatically added to the HTTP spans. Add the attributes and provide support from the exporters.

### PR DESCRIPTION
**Pull request by rakyll**
Fixes #385.

### CODE REVIEW

**Comment by rakyll** [comment]
I will also add a test, please take a look at the APIs for now.

File: exporter/stackdriver/trace_proto.go [hunk 1]
```diff
@@ -163,26 +173,51 @@ func copyAttributes(out **tracepb.Span_Attributes, in map[string]interface{}) {
...
-		case bool:
-			av.Value = &tracepb.AttributeValue_BoolValue{BoolValue: value}
-		case int64:
-			av.Value = &tracepb.AttributeValue_IntValue{IntValue: value}
-		case string:
-			av.Value = &tracepb.AttributeValue_StringValue{StringValue: trunc(value, maxAttributeStringValue)}
+		av := attributeValue(value)
+		switch key {
+		case httptrace.URL:
+			(*out).AttributeMap[labelHTTPURL] = &av
```
Comment (odeke-em) [inline exporter/stackdriver/trace_proto.go]: Looks iffy as we are repeating the pointer creation in every case, perhaps we can reverse this and do it once e.g
```go
func attributeValue(v interface{}) *tracepb.AttributeValue
```

and then we don't have to take the address in every spot.

File: plugin/http/httptrace/httptrace.go [hunk 2]
```diff
@@ -27,6 +27,18 @@ import (
...
 // TODO(jbd): Add godoc examples.
 
+// Attributes recorded on the span for the requests.
+const (
+	URL          = "http.url"
+	Host         = "http.host"
+	Method       = "http.method"
+	Path         = "http.path"
+	UserAgent    = "http.user_agent"
+	StatusCode   = "http.status"
```
Comment (odeke-em) [inline plugin/http/httptrace/httptrace.go]: Where are we getting the enumeration of these attributes?

Comment (odeke-em) [reply plugin/http/httptrace/httptrace.go]: I mention this because "http.status_code", "http.status" could even be different according to Go's HTTP API

**Pull request merged**